tensorflow_recommenders_addons/dynamic_embedding/core/kernels/lookup_impl/lookup_table_op_cpu.h

```diff
@@ -238,6 +247,80 @@ class TableWrapperOptimized final : public TableWrapperBase<K, V> {
     return Status::OK();
   }
 
+  Status save_to_hdfs(OpKernelContext* ctx, int64 value_dim,
+                      const string& filepath,
+                      const size_t buffer_size) override {
+    size_t dim = static_cast<size_t>(value_dim);
+    auto lt = table_->lock_table();
+
+    HadoopFileSystem hdfs;
+    std::unique_ptr<WritableFile> writer;
+    const string tmp_file = filepath + ".tmp";
+    TF_RETURN_IF_ERROR(hdfs.NewWritableFile(tmp_file, &writer));
+
+    uint32 record_len = sizeof(K) + sizeof(V) * dim;
+    uint64 pos = 0;
+    uint8 content[buffer_size + record_len];
+
+    for (auto it = lt.begin(); it != lt.end(); ++it) {
+      K k = it->first;
+      std::memcpy(content + pos, reinterpret_cast<uint8*>(&k), sizeof(K));
+
+      const auto& jt = it->second.data();
+      std::memcpy(content + pos + sizeof(K), reinterpret_cast<uint8*>(jt),
+                  sizeof(V) * dim);
```

Comment:
the sizeof one value : sizeof(V) * dim, as a const variable, could be calculate out of this for loop, for better performance

Reply:
accept